src/Symfony/Component/DependencyInjection/Container.php

```diff
@@ -269,10 +269,10 @@ public function get($id, $invalidBehavior = self::EXCEPTION_ON_INVALID_REFERENCE
                     }
 
                     $alternatives = array();
-                    foreach ($this->services as $key => $associatedService) {
-                        $lev = levenshtein($id, $key);
-                        if ($lev <= strlen($id) / 3 || false !== strpos($key, $id)) {
-                            $alternatives[] = $key;
+                    foreach ($this->getServiceIds(false) as $knownId) {
```

Comment:
this change is unrelated and should be discussed in #19685 (thus no need for the patch here)